docs/architecture/raft.rst

```diff
@@ -4,112 +4,250 @@ Raft Consensus Algorithm in ScyllaDB
      - raft
 
-If you upgrade to ScyllaDB 5.0 from an earlier version, perform a :doc:`rolling restart </operating-scylla/procedures/config-change/rolling-restart/>` 
-updating the ``scylla.yaml`` file for **each node** in the cluster to enable the experimental Raft feature:
+.. _enabling-raft-existing-cluster:
 
-.. code-block:: yaml
-    
-    experimental_features:
-      - raft
+Enabling Raft in ScyllaDB 5.2 and further
+=========================================
+
+.. TODO include enterprise versions in this documentation
 
+.. note::
+  In ScyllaDB 5.2, Raft is Generally Available and can be safely used for consistent schema management.
+  In ScyllaDB 5.3 it will become enabled by default.
+  In further versions it will be mandatory.
 
-When all the nodes in the cluster and updated and restarted, the cluster will begin to use Raft for schema changes.
+ScyllaDB 5.2 and later comes equipped with a procedure that can setup Raft in an existing cluster. We refer to this as the **internal Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/upgrade-opensource/upgrade-guide-from-5.1-to-5.2/upgrade-guide-from-5.1-to-5.2-generic>`).
 
 .. warning::
     Once enabled, Raft cannot be disabled on your cluster. The cluster nodes will fail to restart if you remove the Raft feature.
 
-Verifying that Raft Is Enabled
+To enable Raft in an existing cluster in Scylla 5.2 and beyond:
+
+* ensure that the schema is synchronized in the cluster by executing :doc:`nodetool describecluster </operating-scylla/nodetool-commands/describecluster>` on each node and ensuring that the schema version is the same on all nodes,
+* then perform a :doc:`rolling restart </operating-scylla/procedures/config-change/rolling-restart/>`, updating the ``scylla.yaml`` file for **each node** in the cluster before restarting it to enable the ``consistent_cluster_management`` flag:
+
+.. code-block:: yaml
+
+   consistent_cluster_management: true
+
+When all the nodes in the cluster and updated and restarted, the cluster will start the **internal Raft upgrade procedure**.
+**You must then verify** that the internal Raft upgrade procedure has finished successfully. Refer to the :ref:`next section <verify-raft-procedure>`.
+
+You can also enable the ``consistent_cluster_management`` flag while performing :doc:`rolling upgrade from 5.1 to 5.2 </upgrade/upgrade-opensource/upgrade-guide-from-5.1-to-5.2/upgrade-guide-from-5.1-to-5.2-generic>`: update ``scylla.yaml`` before restarting each node. The internal Raft upgrade procedure will start as soon as the last node was upgraded and restarted. As above, this requires :ref:`verifying <verify-raft-procedure>` that this internal procedure successfully finishes.
+
+Finally, you can enable the ``consistent_cluster_management`` flag when creating a new cluster. This does not use the internal Raft upgrade procedure; instead, Raft is functioning in the cluster and managing schema right from the start.
```

Comment:
To disable this flag once enabled, follow these steps <>
Or explain it can't be disabled.

Reply:
There's a big warning right at the beginning of this "Enabling Raft in ScyllaDB 5.2 and further" section:

```
.. warning::
    Once enabled, Raft cannot be disabled on your cluster. The cluster nodes will fail to restart if you remove the Raft feature.
```


(if you didn't see the documentation rendered - you should)

Reply:
Added this remark at the end:
> Until all nodes are restarted with ``consistent_cluster_management: true``, it is still possible to turn this option back off. Once enabled on every node, it must remain turned on (or the node will refuse to restart).